logcheck/pkg/logcheck.go

```diff
@@ -561,6 +565,14 @@ func kvCheck(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName st
 
 	for index, arg := range keyValues {
 		if index%2 != 0 {
+			// Check values?
```

Comment:
Oh, this makes this loop even harder to read. Could we separate key and value validation by using a dedicated functions? For example 
```
func kvCheck(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string, keyCheckEnabled, parametersCheckEnabled, valueCheckEnabled bool) {
  ...
  for index, arg := range keyValues {
    if index%2 == 0 {
      checkKey(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string, keyCheckEnabled) 
      checkParameters(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string, parametersCheckEnabled) 
    }
    checkValue(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string, valueCheckEnabled) 
  }
}
```

Reply:
Agreed, let's split it up.

I think a switch statement with two cases works best here because it puts both check calls at the same indention. Indenting one but not the other as in an if+continue looks weird.

Reply:
Yes, switch is great idea. I hate golang aversion for `else`.